pkg/client/client.go

```diff
@@ -1526,42 +1528,27 @@ func (c *Client) CreateOrUpdateClusterRoleBinding(ctx context.Context, crb *rbac
 
 func (c *Client) CreateOrUpdateServiceAccount(ctx context.Context, sa *v1.ServiceAccount) error {
 	sClient := c.kclient.CoreV1().ServiceAccounts(sa.GetNamespace())
-	_, err := sClient.Get(ctx, sa.GetName(), metav1.GetOptions{})
+	existing, err := sClient.Get(ctx, sa.GetName(), metav1.GetOptions{})
 	if apierrors.IsNotFound(err) {
 		_, err := sClient.Create(ctx, sa, metav1.CreateOptions{})
 		return errors.Wrap(err, "creating ServiceAccount object failed")
```

Comment:
Shouldn't we be checking the new value of `err` instead of always assuming creation failed?
And then we should assign the new client to `existing`, to be able to use it later

Reply:
It's in this case we don't need to check it since `Wrap` already does it for us ;)

> Wrap returns an error annotating err with a stack trace
> at the point Wrap is called, and the supplied message.
> If err is nil, Wrap returns nil.